cdisc_rules_engine/utilities/excel_report.py

```diff
@@ -162,3 +51,21 @@ def get_excel_export(self, define_version, cdiscCt, standard, version) -> Workbo
         wb["Conformance Details"]["B10"] = ", ".join(cdiscCt)
         wb["Conformance Details"]["B11"] = define_version
         return wb
+
+    def write_report(self):
+        output_name = self._args.output + "." + self._args.output_format.lower()
+        logger = logging.getLogger("validator")
+        try:
+            report_data = self.get_export(
+                self._args.define_version,
+                self._args.controlled_terminology_package,
+                self._args.standard,
+                self._args.version.replace("-", "."),
+            )
+            with open(output_name, "wb") as f:
+                f.write(excel_workbook_to_stream(report_data))
+        except Exception as e:
+            logger.error(e)
+            raise e
+        finally:
+            self._template.close()
```

Comment:
It might be a bad design, but it is opened in ReportFactory (line 41 - self._data_service.read_data(..)). I kept it this way for backward compatibility with tests, but I can rewrite ti, so that the template is opened within a class instance and not passed as a parameter.